### PR TITLE
component usages filters

### PR DESCRIPTION
**Pull request by JulianWielga**
upgraded usages display
added filter for usage type
fixed text filter
cleanup

### CODE REVIEW

File: designer/submodules/packages/components/src/components/usages/componentView.tsx [hunk 1]
```diff
@@ -54,6 +55,14 @@ function Component(): JSX.Element {
                             return t("table.filter.SCENARIOS", "Scenarios");
                     }
                     break;
+                case "USAGE_TYPE":
+                    switch (value) {
+                        case UsagesFiltersUsageType.INDIRECT:
+                            return t("table.filter.INDIRECT", "Indirect usage");
+                        case UsagesFiltersUsageType.STRAIGHT:
+                            return t("table.filter.STRAIGHT", "Straight usage");
```
Comment (arkadius) [inline designer/submodules/packages/components/src/components/usages/componentView.tsx:63]: Straight is the correct word for that? I think "direct" sounds better. Let's keep this simple.

File: designer/submodules/packages/components/src/components/usages/componentView.tsx [hunk 2]
```diff
@@ -67,11 +76,32 @@ function Component(): JSX.Element {
...
+            switch (type) {
+                case "USAGE_TYPE":
+                    switch (value) {
+                        case UsagesFiltersUsageType.INDIRECT:
+                            return theme.palette.secondary.main;
+                        case UsagesFiltersUsageType.STRAIGHT:
+                            return theme.palette.primary.main;
+                    }
+            }
+            return getRandomColorByType(type);
```
Comment (arkadius) [inline designer/submodules/packages/components/src/components/usages/componentView.tsx:91]: Can you write a comment why here is picked a random color? I know that it was done this way before your change, but now it is more explicit and I'm wondering why this is done that way

File: designer/submodules/packages/components/src/components/usages/nodesCell.tsx [hunk 3]
```diff
@@ -1,70 +1,106 @@
...
 import { NodeUsageData } from "nussknackerUi/HttpService";
+import { UsageWithStatus } from "../useComponentsQuery";
+import { UsagesFiltersModel } from "./usagesFiltersModel";
+import { nodeFilter } from "./usagesTable";
 
 const icon = <LinkIcon />;
 
-export const NodesCell = ({ filterText, ...props }: GridRenderCellParams & { filterText: string }): JSX.Element => {
+export function getNodeName({ fragmentNodeId, nodeId }: Pick<NodeUsageData, "fragmentNodeId" | "nodeId">): string {
+    return fragmentNodeId ? `${nodeId} ∈ ${fragmentNodeId}` : nodeId;
```
Comment (arkadius) [inline designer/submodules/packages/components/src/components/usages/nodesCell.tsx:15]: ∈ ?

File: designer/submodules/packages/components/src/components/usages/componentView.tsx (hunk 1 above)
Comment (jzdr) [reply designer/submodules/packages/components/src/components/usages/componentView.tsx:63]: Agree - direct/indirect would fit better here.

File: designer/submodules/packages/components/src/components/usages/componentView.tsx (hunk 2 above)
Comment (JulianWielga) [reply designer/submodules/packages/components/src/components/usages/componentView.tsx]: its not random, based on string provided

File: designer/submodules/packages/components/src/components/usages/componentView.tsx [hunk 4]
```diff
@@ -54,6 +55,14 @@ function Component(): JSX.Element {
                             return t("table.filter.SCENARIOS", "Scenarios");
                     }
                     break;
+                case "USAGE_TYPE":
+                    switch (value) {
+                        case UsagesFiltersUsageType.INDIRECT:
+                            return t("table.filter.INDIRECT", "Indirect usage");
+                        case UsagesFiltersUsageType.STRAIGHT:
```
Comment (arkadius) [inline designer/submodules/packages/components/src/components/usages/componentView.tsx:62]: please change to DIRECT

**Review by arkadius — APPROVED** [review]
I don't like this new `${nodeId} ❮${fragmentNodeId}❯` notation, but let's merge it - other changes are nice

**Pull request merged**